axelrod/strategy_transformers.py

```diff
@@ -228,11 +297,21 @@ def dual_wrapper(player, opponent, proposed_action):
     -------
     action: an axelrod.Action, C or D
     """
+
     if not player.history:
-        player.original_player = player.original_class(**player.init_kwargs)
+        player.use_history = []
+
+    temp = player.history[:]
+    player.history = player.use_history[:]
+
```

Comment:
i'm in the middle of refactoring, but perhaps the new `dual_wrapper` code (doc_str edited out) will make things more clear.

```
def flip_wrapper(player, opponent, proposed_action):
    flip_play_attributes(player)

    if is_strategy_static(player.original_class):
        action = player.original_class.strategy(opponent)
    else:
        action = player.original_class.strategy(player, opponent)

    flip_play_attributes(player)

    return action.flip()
```

`flip_play_attributes` flips `history`, `cooperations`, `defections` and `state_distribution`